gameMechanics.js

```diff
@@ -0,0 +1,303 @@
+// Game Mechanics for the Connect Four Game
+
+// Main methods
+
+var createGrid = function(height, width) {
+	var grid = new Array(width);
+	for (column=0; column < width; column++) {
+		grid[column]=new Array(height);
+	}
+	for (column=0; column < width; column++) {
+		for (row=0; row < height; row++) {
+			grid[column][row] = "<font color='gray'>E</font>";
+		}
+	}
+	return grid;
+}
+
+var printGrid = function(inputGrid, gridName) {
+	var height = inputGrid[0].length;
+	var width = inputGrid.length;
+	var token = "<font size='5'><pre>";
+	var rowIndex = 0;
+	for (row=0; row < height; row++) {
+		for (column=0; column < width; column++) {
+			if (rowIndex == width) {
+				token += "<br><br>";
+				rowIndex = 0;
+			}
+			token += inputGrid[column][row] + " ";
+			rowIndex++;
+		}
+	}
+	token += "</pre></font>";
+	document.getElementById(gridName).innerHTML = token;
+}
+
+var chooseButton = function (currentGrid, gridName, selectedColumn, playerFlag) {
+	var newGrid = insertDisc(currentGrid, selectedColumn, playerFlag);
+	printGrid(newGrid, gridName);
+	printTurn(turn);
+}
+
+var insertDisc = function(currentGrid, selectedColumn, currentPlayer) {
+	var height = currentGrid[selectedColumn].length;
+	var width = currentGrid.length;
+	var position = null;
+	var isEmpty = "<font color='gray'>E</font>";
+	var playerOne = "X";
+    var playerTwo = "<font color='red'>O</font>";
+	var vacantRow = true;
+	for (currentRow=0; vacantRow == true && currentRow < height; currentRow++) {
+		if (currentGrid[selectedColumn][currentRow] == isEmpty) {
+			position = currentRow;
+		}
+		else {
+			vacantRow = false;
+		}
+	}
+	if (position == null && vacantRow == false) {
+		alert("This column is already filled, please try another column.");
+	}
+	else {
+		if (turn % 2 == 0) {
```

Comment:
Ah whoops, I was being redundant and could've used checkPlayer function, which is pretty much that.  I'll rename the function as well.